new-docs/content/any/getting-started/policies/index.md

```diff
@@ -9,325 +9,438 @@ description: Learn about writing Oso policies - the source of truth for authoriz
+would yield values that satisfy the conditions. If the call were,
+however, something like:
 
-## Application Types
+```python
+oso.is_allowed(User(id=1, admin=False), "read", Resource(owner=0))
+```
 
-Any type defined in an application can be passed into Oso, and its attributes
-may be accessed from within a policy. Using application types make it possible
-to take advantage of an app’s existing domain model. For example:
+This call would return `False`, since neither rule would match
+the supplied instances. The first rule would fail because the
+`admin` flag isn't true, and the second because the user's ID
+`1` isn't equal to the resource's owner ID `0`.
 
-{{< code file="policy.polar" >}}
-allow(actor, action, resource) if actor.{{% exampleGet "isAdmin" %}};
-{{< /code >}}
+## Classes
 
-<!-- TODO(gj): Link `Oso.isAllowed()` once API docs are setup. -->
+There's a potential problem with the rules above: they refer to fields
+of objects without checking the _types_ of those objects. For example,
+if we were to pass strings again instead of instances, we'd get an error
+at query time:
 
-The above rule expects the `actor` variable to be a {{% exampleGet "langName"
-%}} {{% exampleGet "instance" %}} with the field `{{% exampleGet "isAdmin" %}}`. The {{% exampleGet "langName" %}} {{% exampleGet "instance" %}} is passed
-into Oso with a call to `Oso.{{% exampleGet "isAllowed" %}}()`:
+```console
+query> allow("Johann", "read", "resource")
+PolarRuntimeError
+...
+Application error: 'str' object has no attribute 'admin'
+```
 
-{{% exampleGet "userClass" %}}
+To guard against this, we can add explicit type checks using
+the Polar `matches` operator:
 
-The code above provides a `User` object as the _actor_ for our `allow` rule.
-Since `User` has a field called `{{% exampleGet "isAdmin" %}}`, it is checked
-during evaluation of the Polar rule and found to be true.
+```polar
+allow(user, "read", resource) if
+    actor matches User and
+    user.admin = true;
+
+allow(user, "read", resource) if
+    user matches User and
+    resource matches Resource and
+    user.id = resource.owner;
+```
 
-In addition to accessing attributes, you can also call methods on application
-instances in a policy:
+The `matches` operator succeeds when its left-hand side is an
+instance of (a subclass of) its right-hand side. For our example
+query above, since `str` is not a subclass of either `User`
+or `Resource`, the `matches` would fail, and not raise an error.
 
-{{< code file="policy.polar" >}}
-allow(actor, action, resource) if actor.{{% exampleGet "isAdminOf" %}}(resource);
-{{< /code >}}
+Here `User` and `Resource` refer to the application classes defined
+above. But in order for Polar to use a class in a type check, it must
+first be _registered_. If you're using an ORM adapter, this happens
+automatically for all model classes; otherwise, you can register
+classes manually using `Oso.register_class`. If you forget, Polar
```

Comment:
Link to API docs here (but this will be language-specific, so adds a lot of overhead). I would do this last if time.